tests/Kiota.Builder.Tests/Writers/Java/CodeMethodWriterTests.cs

```diff
@@ -0,0 +1,306 @@
+            Assert.Contains("writeStringValue", result);
+            Assert.Contains("writeCollectionOfPrimitiveValues", result);
+            Assert.Contains("writeCollectionOfObjectValues", result);
+            Assert.Contains("writeEnumValue", result);
+            Assert.Contains("writeAdditionalData(this.additionalData);", result);
+        }
+        [Fact]
+        public void WritesMethodAsyncDescription() {
+            
+            method.Description = methodDescription;
+            var parameter = new CodeParameter(method){
+                Description = paramDescription,
+                Name = paramName
+            };
+            parameter.Type = new CodeType(parameter) {
+                Name = "string"
+            };
+            method.AddParameter(parameter);
+            writer.Write(method);
+            var result = tw.ToString();
+            Assert.Contains("/**", result);
+            Assert.Contains(methodDescription, result);
+            Assert.Contains("@param ", result);
+            Assert.Contains(paramName, result);
+            Assert.Contains(paramDescription, result); 
+            Assert.Contains("@return a CompletableFuture of", result);
+            Assert.Contains("*/", result);
+        }
+        [Fact]
+        public void WritesMethodSyncDescription() {
+            
+            method.Description = methodDescription;
+            method.IsAsync = false;
+            var parameter = new CodeParameter(method){
+                Description = paramDescription,
+                Name = paramName
+            };
+            parameter.Type = new CodeType(parameter) {
+                Name = "string"
+            };
+            method.AddParameter(parameter);
+            writer.Write(method);
+            var result = tw.ToString();
+            Assert.DoesNotContain("@return a CompletableFuture of", result);
+        }
+        [Fact]
+        public void Defensive() {
+            var codeMethodWriter = new CodeMethodWriter(new JavaConventionService());
+            Assert.Throws<ArgumentNullException>(() => codeMethodWriter.WriteCodeElement(null, writer));
+            Assert.Throws<ArgumentNullException>(() => codeMethodWriter.WriteCodeElement(method, null));
+        }
+        [Fact]
+        public void ThrowsIfParentIsNotClass() {
+            method.Parent = CodeNamespace.InitRootNamespace();
+            Assert.Throws<InvalidOperationException>(() => writer.Write(method));
+        }
+        [Fact]
+        public void ThrowsIfReturnTypeIsMissing() {
+            method.ReturnType = null;
+            Assert.Throws<InvalidOperationException>(() => writer.Write(method));
+        }
+        [Fact]
+        public void WritesReturnType() {
+            writer.Write(method);
+            var result = tw.ToString();
+            Assert.Contains(methodName, result);
+            Assert.Contains(returnTypeName, result);
+            Assert.Contains("CompletableFuture<", result);// async default
```

Comment:
Opportunity to test the format if you combine these assertions to test the entire string at once.

Reply:
done